lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -2584,37 +2586,104 @@ lldb::addr_t SwiftLanguageRuntime::GetAsyncContext(RegisterContext *regctx) {
+    return ptr;
+  return {};
+}
+
+/// Converts the abstract location of UnwindPlan::Row::FAValue into a concrete
+/// address. This is a simplified version of the methods in
+/// RegisterContextUnwind, since plumbing access to those here would be
+/// challenging.
+std::optional<addr_t> GetCFAAddr(Process &process, RegisterContext &regctx,
+                                 UnwindPlan::Row::FAValue cfa_loc) {
+  if (cfa_loc.IsRegisterPlusOffset()) {
+    unsigned regnum = cfa_loc.GetRegisterNumber();
+    if (std::optional<addr_t> regvalue = ReadRegisterAsAddress(regctx, regnum))
+      return *regvalue + cfa_loc.GetOffset();
+  }
+  return {};
+}
+
+/// Attempts to use UnwindPlans that inspect assembly to recover the entry value
+/// of the async register. This is a simplified version of the methods in
+/// RegisterContextUnwind, since plumbing access to those here would be
+/// challenging.
+std::optional<addr_t>
+ReadAsyncRegFromUnwind(SymbolContext &sc, Process &process, Address pc,
+                       Address func_start_addr, RegisterContext &regctx,
+                       AsyncUnwindRegisterNumbers regnums) {
+  FuncUnwindersSP unwinders =
+      pc.GetModule()->GetUnwindTable().GetFuncUnwindersContainingAddress(pc,
+                                                                         sc);
+  if (!unwinders)
+    return {};
 
-  const uint8_t *expr = triple == llvm::Triple::x86_64
-                            ? g_cfa_dwarf_expression_x86_64
-                            : g_cfa_dwarf_expression_arm64;
-  auto size = triple == llvm::Triple::x86_64
-                  ? sizeof(g_cfa_dwarf_expression_x86_64)
-                  : sizeof(g_cfa_dwarf_expression_arm64);
-  if (with_double_deref)
-    return llvm::ArrayRef<uint8_t>(expr, size);
-  return llvm::ArrayRef<uint8_t>(expr, size - 1);
+  Target &target = process.GetTarget();
+  UnwindPlanSP unwind_plan =
+      unwinders->GetUnwindPlanAtNonCallSite(target, regctx.GetThread());
+  if (!unwind_plan)
+    return {};
+
+  UnwindPlan::RowSP row = unwind_plan->GetRowForFunctionOffset(
+      pc.GetFileAddress() - func_start_addr.GetFileAddress());
+  UnwindPlan::Row::AbstractRegisterLocation regloc;
+
+  // If the plan doesn't have information about the async register, we can use
+  // its current value, as this is a callee saved register.
+  if (!row->GetRegisterInfo(regnums.async_ctx_regnum, regloc))
+    return ReadRegisterAsAddress(regctx, regnums.async_ctx_regnum);
+
+  // Handle the few abstract locations we are likely to encounter.
+  using RestoreType = UnwindPlan::Row::AbstractRegisterLocation::RestoreType;
+  RestoreType loctype = regloc.GetLocationType();
+  switch (loctype) {
+  case RestoreType::same:
+  case RestoreType::inOtherRegister: {
+    unsigned regnum = loctype == RestoreType::same ? regnums.async_ctx_regnum
+                                                   : regloc.GetRegisterNumber();
+    return ReadRegisterAsAddress(regctx, regnum);
```

Comment:
Another place where you need to convert a register number into the `eRegisterKindLLDB` domain.